test/tests/amqp/test_amqp_racknode.py

```diff
@@ -0,0 +1,166 @@
+                "Incorrect HTTP return code: {}".format(mon_data['status']))
+            json_node_data = mon_data['json']
+            self.assertTrue(
+                json_node_data['name'] == newrack['name'] and json_node_data['type'] == "rack",
+                "rack node field error")
+            timecount = 0
+            while amqp_message_received is False and timecount < 10:
+                sleep(1)
+                timecount = timecount + 1
+            self.assertNotEquals(timecount, 10, "No AMQP message received")
+            expectedkey = "node.added.information." + rackid + '.' + rackid
+            expectedpayload = {"type": "node",
+                               "action": "added",
+                               "typeId": rackid,
+                               "nodeId": rackid,
+                               "severity": "information",
+                               "version": "1.0",
+                               "createdAt": mon_data_post['json']['createdAt']}
+            self.assertEquals(self.compare_message(expectedkey, expectedpayload), True, "AMQP Message Check Error!")
+            if fit_common.VERBOSITY >= 2:
+                print "query rack: " + rackname + "successfully!"
+        if fit_common.VERBOSITY >= 2:
+            print "test: rack creation and query succeed!"
+
+    def test_api_delete_rack(self):
+        rack_node_list = get_node_list_by_type("rack")
+        for rack in rack_node_list:
+            global amqp_message_received
+            amqp_message_received = False
+            # start amqp thread
+            td = fit_amqp.AMQP_worker(
+                exchange_name="on.events", routing_key="node.removed.#",
+                externalcallback=amqpcallback, timeout=10)
+            td.setDaemon(True)
+            td.start()
+            mon_url = '/api/2.0/nodes/{}'.format(rack)
+            mon_data = fit_common.rackhdapi(mon_url, action='delete')
+            self.assertIn(
+                mon_data['status'], range(200, 205),
+                "Incorrect HTTP return code: {}".format(mon_data['status']))
+            timecount = 0
+            while amqp_message_received is False and timecount < 10:
+                sleep(1)
+                timecount = timecount + 1
+            self.assertNotEquals(timecount, 10, "No AMQP message received")
+            expectedkey = "node.removed.information." + rack + '.' + rack
+            expectedpayload = {"type": "node",
+                               "action": "removed",
+                               "typeId": rack,
+                               "nodeId": rack,
+                               "severity": "information",
+                               "version": "1.0",
+                               "createdAt": mon_data['headers']['Date']}
+            self.assertEquals(self.compare_message(expectedkey, expectedpayload), True, "AMQP Message Check Error!")
+
+
+if __name__ == '__main__':
+    fit_common.unittest.main()

```

Comment:
Use this instead: 
if __name__ == '__main__':
    fit_common.run_from_module(__file__)